### ISSUE TITLE
Station will break down when using ECO.

### ISSUE THREAD

**Issue by gogochlwns2** [DeadAirRT/deadair_eco#9]
![2024-12-21 11 11 57](https://github.com/user-attachments/assets/9b302e72-cf70-452b-8b9a-3b36d2ce5a2a)

The first symptom is that the plot management table of the station construction simulation stops working.
When trying to interact with it, the window does not open.

The second symptom is that the turrets on the station stop working. They do not work under any circumstances. Also, when you try to select the plot management window of a station with a broken turret, the window closes immediately.

The second symptom does not occur immediately, but randomly during play.

The error logs for the first and second symptoms are the same.

[](https://pastebin.com/tGx9ZqsR)

[=ERROR=] 2.91 CWidgetController::UpdateFrame() - Validation error: 'Script error: 'Invalid cell content [row: 6, col: 1]. Start value is bigger than max select value.''
[=ERROR=] 2.91 C:/Program Files (x86)/Steam/steamapps/common/X4 Foundations/.../viewhelper.lua(63): UpdateFrame(): UpdateFrame(): Failed to set up the frame.
[=ERROR=] 2.91 stack traceback:
	[C]: in function 'TraceBack'
	P:\SVN\temp\helper.lua:3912: in function 'callback'
	P:\SVN\temp\viewhelper.lua:70: in function 'updateMenu'
	P:\SVN\temp\viewhelper.lua:167: in function 'registerMenu'
	P:\SVN\temp\helper.lua:3762: in function 'display'
	P:\SVN\temp\menu_map.lua:6313: in function 'createInfoFrame'
	P:\SVN\temp\menu_map.lua:6123: in function 'displayMenu'
	P:\SVN\temp\menu_map.lua:5673: in function 'onShowMenu'
	P:\SVN\temp\helper.lua:1248: in function <P:\SVN\temp\helper.lua:1151>
	[C]: in function 'pcall'
	P:\SVN\temp\widget_fullscreen.lua:17931: in function 'CallEventScripts'
	P:\SVN\temp\widget_fullscreen.lua:2342: in function <P:\SVN\temp\widget_fullscreen.lua:2325>
[=ERROR=] 2.91 Error while executing onEvent script for event: showMapMenu.
Errormessage: P:\SVN\temp\helper.lua:3914: Failed to create view for menu MapMenu. Reloading UI ...

**Comment by DeadAirRT** [comment]
First of all, thank you for the detailed report.

This change only affects AI default plot sizes to be larger and should not have any effect on player station plots in real games. Likely has to do with however egosoft set up the station planner. I'll see if I can't find a solution for that.

The turret problems are completely out of the scope of any of my changes and I've yet to ever experience it or receive any other reports. I'll keep an eye out for it, but I believe it's more likely a vanilla bug.

**Comment by gogochlwns2** [comment]
You seem to be right.

The plot selection issue has been fixed, but the random turret failure issue is still occurring.

When I asked about it on Reddit, someone reported the same symptoms, so it seems to be a vanilla issue.

**Comment by DeadAirRT** [comment]
I adjusted the default plot size for x4 v7.5. Let me know if anyone is experiencing issues still.

**Comment by Vectorial1024** [comment]
It seems there are UI problems reported. UI problems can cause UI screens to fail to load, i.e. when there should be a screen, there is no screen and the player is just looking directly to the game-rendered universe with unresponsive buttons.

Looking at your mod list, I notice `sn_mod_support_apis` (SirNuke's Mod Support API), which is a well-known UI mod. It is likely this UI mod is causing problems, but not DeadAir_Eco itself.

Some views:
- Perhaps this is already fixed by SirNuke's side (please go ask on their side!)
- X4 Foundations 7.5 (released in Dec 2024) introduces a new feature "protected UI", which basically blocks UI mods from loading
  - This matches the timing of OP's post
  - If no other mods depend on SirNuke's, then SirNuke's mod should be removed
  - If SirNuke's mod is really needed, then disable the game's protected UI mode

Bottom line: it just seems very unlikely for this problem to be related to DeadAir_Eco.
